pkg/certificate/manager.go

```diff
@@ -16,30 +17,84 @@ var (
 	errCertNotFound = errors.New("certificate not found")
 )
 
+const checkCertificateExpirationInterval = 5 * time.Second
+
 // NewManager creates a new CertManager with the passed CA and CA Private Key
 func NewManager(
 	ca *Certificate,
 	client client,
 	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker) (*manager, error) { //nolint:revive // unexported-return
+	msgBroker *messaging.Broker) (*Manager, error) {
 	if ca == nil {
 		return nil, errNoIssuingCA
 	}
 
-	m := &manager{
+	m := &Manager{
 		// The root certificate signing all newly issued certificates
 		ca:                          ca,
 		client:                      client,
 		serviceCertValidityDuration: serviceCertValidityDuration,
 		msgBroker:                   msgBroker,
 	}
+	m.start(checkCertificateExpirationInterval)
+	return m, nil
+}
 
-	// TODO(#4533) start the cert rotation here.
+func (m *Manager) start(checkInterval time.Duration) {
+	// iterate over the list of certificates
+	// when a cert needs to be rotated - call RotateCertificate()
+	ticker := time.NewTicker(checkInterval)
+	done := make(chan bool)
+	m.stopRotation = done
```

Comment:
I don't think this is doing quite what you want. I would expect this channel to already exist and be passed in to `NewManager` vs. being created in this function. Then the caller can `close` the channel when it wants this to stop. Otherwise I don't see where this is can be triggered to stop besides the uncalled `stop()` method.